### PR TITLE
Add MQ User resource v5.31.0

### PR DESCRIPTION
**Pull request by mergenci**
### Description

[MQ User](https://docs.aws.amazon.com/amazon-mq/latest/api-reference/brokers-broker-id-users-username.html) resource is currently managed [inside MQ Broker](https://github.com/upbound/terraform-provider-aws/blob/upjet-v5.31.0/internal/service/mq/broker.go#L297-L344) resource. This PR implements MQ User as an independent resource. Note that a Broker restart is required for any MQ User operations to take effect.

### References

- On how to introduce a new Terraform resource, see https://github.com/upbound/terraform-provider-aws/pull/132
- Also see PR for v5.50.0: https://github.com/upbound/terraform-provider-aws/pull/245

### CODE REVIEW

File: tools/tfsdk2fw/go.mod
Comment (ulucinar) [inline tools/tfsdk2fw/go.mod:1]: Why do we have changes in this file?

File: .ci/tools/go.mod
Comment (ulucinar) [inline .ci/tools/go.mod:1]: Can we avoid these changes?

File: go.mod
Comment (ulucinar) [inline go.mod:1]: I believe these are side effects from the replacing the base. We had better avoid any changes in the dependencies.

**Review by ulucinar — COMMENTED** [review]
Thank you @mergenci for introducing the `aws_mq_user` resource. We had better avoid any dependency changes in the go.mod files. Was this a side effect of changing the base?

File: go.mod
Comment (mergenci) [reply go.mod:1]: For posterity, Alper's comments addressed changes introduced after switching merge target branch from upjet-v5.31.0 to upjet-v5.50.0. We reverted back to upjet-v5.31.0.

File: internal/service/mq/user.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource(name="User")
```
Comment (erhancagirici) [inline internal/service/mq/user.go:32]: kudos for implementing it as a framework resource 👍

File: internal/service/mq/user.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	userDetails, err := findUserByID(ctx, conn, state.BrokerID.ValueString(), state.ID.ValueString())
+	if tfresource.NotFound(err) {
+		create.LogNotFoundRemoveState(names.MQ, create.ErrActionReading, ResourceNameUser, fmt.Sprintf("%s/%s", state.BrokerID.ValueString(), state.ID.ValueString()))
+		response.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		response.Diagnostics.Append(create.DiagErrorFramework(names.MQ, create.ErrActionReading, ResourceNameUser, fmt.Sprintf("%s/%s", state.BrokerID.ValueString(), state.ID.ValueString()), err))
+		return
+	}
```
Comment (erhancagirici) [inline internal/service/mq/user.go:141]: Note: This is more related to the XP AWS provider side, but wanted to mention here anyways.
AFAIU, the User's `id` is available after the creation. Does it match the `username` or a generated one by the AWS API ?

At the very start of the MR creation, we start with an empty id at MR manifest.
The external name configuration should be able to handle the first external observation before external create, so that it should return "not found" without any TF error diagnostics. Otherwise, the first observe will fall into this error path (due to empty ID in the AWS SDK call), and always get re-queued at XP managed reconciler.
TLDR: XP external name config should be done appropriately, and ensure first Observe should receive "Not Found" from downstream TF Read call.

Comment (erhancagirici) [reply internal/service/mq/user.go:141]: By looking at the other parts, it looks like the ID is the same with username, so we have a deterministic TF ID 👍 Please ignore my above comment

**Review by erhancagirici — APPROVED** [review]
@mergenci many thanks for the great work. This also serves as a good reference for new TF resource implementations. LGTM!
A small but important note is: in a follow-up, it would be nice to document/automate this process for version bumps.

**Comment by mergenci** [comment]
Thanks for your reviews @ulucinar and @erhancagirici 🙏
